src/conversations/Conversations.ts

```diff
@@ -429,6 +439,72 @@ export default class Conversations {
     return seenPeers
   }
 
+  async newGroupConversation(
+    initialMembers: string[]
+  ): Promise<GroupConversation> {
+    if (initialMembers.length === 0) {
+      throw new Error('No initial members provided')
+    }
+
+    const groupID = bytesToHex(crypto.getRandomValues(new Uint8Array(32)))
+    const context = {
+      conversationId: `xmtp.org/groups/${groupID}`,
+      metadata: {
+        initialMembers: initialMembers.join(','),
+      },
+    }
+
+    const timestamp = new Date()
+    const members = await Promise.all(
+      initialMembers.map(async (member) => {
+        let contact = await this.client.getUserContact(member)
+        if (!contact) {
+          throw new Error(`Recipient ${member} is not on the XMTP network`)
+        }
+
+        // Coerce the contact into a V2 bundle
+        if (contact instanceof PublicKeyBundle) {
+          contact = SignedPublicKeyBundle.fromLegacyBundle(contact)
+        }
+
+        return toSignedPublicKeyBundle(contact)
+      })
+    )
+
+    const inviteResponses = await this.client.keystore.createInvites({
+      recipients: members,
+      context,
+      createdNs: dateToNs(timestamp),
+    })
+
+    const envelopes = inviteResponses.map((response) => {
+      if (!response.conversation) {
+        throw new Error(
+          'no conversation for response: ' + JSON.stringify(response)
+        )
+      }
+
+      return {
+        contentTopic: buildUserInviteTopic(response.conversation?.peerAddress),
+        message: response.payload,
+        timestamp,
+      }
+    })
+
+    await this.client.publishEnvelopes(envelopes)
```

Comment:
I'm not seeing where we invite ourselves to the group. Maybe I'm missing it

Reply:
I’ve been passing the creator’s address inside the initialMembers array, I could see just adding it here and deduping (which we should probably do anyway). Good call.

Reply:
Fixed in https://github.com/xmtp/xmtp-js/pull/381/commits/748720e486376536ef3e336ddba07440ad50f99c.

Reply:
Ideally we would just copy one of the other messages into our own invite topic. Then we don't have to worry about the security implications of self diffie Hellman.

Reply:
Aren't the sealed invitations DH'd anyway? I don't know if we can avoid self DH if we're sending the invites from the recipient?

Reply:
Sort of wondering if the shared wallet approach would actually be helpful due to this..

Reply:
Fixed in https://github.com/xmtp/xmtp-js/pull/381/commits/1746cca16a1545059ff81395320530ed1b9731d2.